docs/pages/world/resource-ids.mdx

```diff
@@ -0,0 +1,106 @@
+  Namespaces are used for [access control](/world/namespaces-access-control).
+- [System (`sy`)](https://github.com/latticexyz/mud/blob/main/packages/world/src/worldResourceTypes.sol#L25).
+  A [system](./systems) contains logic that interact with table data onchain.
+- [Module (`md`)](https://github.com/latticexyz/mud/blob/main/packages/world/src/worldResourceTypes.sol#L21).
+  A module is a deployed script that can install tables, systems, etc.
+
+## Creating resource identifiers
+
+To create a resource ID in your Solidity code you can use `WorldResourceIdLib.encode`.
+This function accepts a single parameter, a structure with three string fields:
+
+- `typeId` is the resource type.
+- `namespace` is the namespace in which the resource is located.
+  If this is the root namespace, use the empty string.
+- `name` is the name of the resource, except in the case of namespaces, where it is empty.
+
+For example, in the code below we create a resource for a table in the root namespace whose name is `Tasks`.
+
+<CollapseCode>
+
+```solidity filename="DeployKeyWithValueModule.s.sol" copy showLineNumbers {7-9,20-24}
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.21;
+import { Script } from "forge-std/Script.sol";
+import { console } from "forge-std/console.sol";
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+import { IWorld } from "../src/codegen/world/IWorld.sol";
+import { WorldResourceIdLib, WorldResourceIdInstance } from "@latticexyz/world/src/WorldResourceId.sol";
+import { RESOURCE_TABLE } from "@latticexyz/world/src/worldResourceTypes.sol";
+import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
+import { KeysWithValueModule } from "@latticexyz/world-modules/src/modules/keyswithvalue/KeysWithValueModule.sol";
+
+contract DeployKeyWithValueModule is Script {
+  function run() external {
+    uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+    address worldAddress = vm.envAddress("WORLD_ADDRESS");
+    vm.startBroadcast(deployerPrivateKey);
+    IWorld world = IWorld(worldAddress);
+    StoreSwitch.setStoreAddress(worldAddress);
+    // Deploy the module
+    KeysWithValueModule keysWithValueModule = new KeysWithValueModule();
+    ResourceId sourceTableId = WorldResourceIdLib.encode({ typeId: RESOURCE_TABLE, namespace: "", name: "Tasks" });
+    world.installRootModule(keysWithValueModule, abi.encode(sourceTableId));
+    vm.stopBroadcast();
+  }
+}
+```
+
+</CollapseCode>
+
+## Understanding resource identifiers
+
+Resource identifiers are usually accessible at 32-byte values in hexadecimal.
+For example, in the MUD Dev Tools, when you see a value such as `0x7462...0000`, you can copy it and paste it somewhere else to see the complete value.
+
+### Online calculator
+
+You can use [an online calculator](https://www.duplichecker.com/hex-to-text.php) to convert hex to text.
+The weird `�` characters you see are where the value of a byte is zero.
+
+### JavaScript
```

Comment:
Can we add a section for typescript with our libraries here in addition? (I still think the "JavaScript" section is useful for a quick and dirty snippet folks can copy into their browser console, but would recommend people to use the ts library whenever they're in a MUD project.)

### TypeScript

```typescript
import { hexToResource } from "@latticexyz/common";

const resource = hexToResource(`0x746273746f72650000000000000000005461626c657300000000000000000000n`);
```